src/main/scala/java/util/logging/Logger.scala

```diff
@@ -107,8 +107,9 @@ class Logger(name: String, resourceBundle: String) {
     if (isLoggable(record.getLevel)) {
       if (useParentsHandlers) {
         val parent = getParent()
-        if (parent != null)
-          parent.log(record)
+        if (parent != null) {
+          parent.getHandlers.foreach(_.publish(record))
+        }
```

Comment:
This will ignore grand-parents that would have registered handlers.

Looks like we need a `private def publish(record): Unit` in `Logger`, that would contain everything in the `{}` of `if (isLoggable(...))`, and recursively call `parent.publish(record)` rather than looking inside `parent.getHandlers`.